database_normalization/database_normalization.md

```diff
@@ -0,0 +1,370 @@
+
+**order_items**
+
+*The second column here is a foreign key.  Sometimes (but not always!) you will see this noted in database diagrams as (FK) or a symbol like <span class="fa-solid fa-key"></span>.*
+
+<!-- data-type="none" class="tight-table" -->
+| order_id | item_id (FK) |
+| :--------- | :--------- |
+| 23125 | 210 |
+| 23125 | 108 |
+| 41320 | 97 |
+| 41320 | 233 |
+| 41320 | 210  |
+| 53011 | 178 |
+| 53011 | 15 |
+| 14123 | 108 |
+| 14123 | 210 |
+| 14123 | 97 |
+
+
+### Quiz: Keys
+
+Consider the following fictional snippet of data from a table:
+
+**medication\_order**
+
+<!-- data-type="none" -->
+| med\_ord\_id   | patient\_id   | date  | provider\_id  |
+| :--------- | :--------- | :--------- | :--------- |
+| 3411904326   | 834014     | 2020-01-30    | 56024     |
+| 3411904329   | 320092     | 2020-01-30    | 56024     |
+| 3411904330   | 612311     | 2020-01-31    | 99123     |
+
+Which of the fields in this table is likely to be a primary key?  Select all that apply.
+
+[[X]] med\_ord\_id
+[[ ]] patient\_id
+[[ ]] date
+[[ ]] provider\_id
+*********
+
+<div class = "answer">
+
+Because this table is about medication orders, and there's an identifier that looks like it uniquely identifies each of these interactions, we can guess that "med\_ord\_id" is probably a primary key.  On the other hand, we know for sure that "date" and "provider\_id" can't be primary keys, because they have repeating values.  That makes sense -- the same provider can write many different medication orders, and on any given date, there are likely to be many medication orders.  By that same logic, we also know that "patient\_id" can't be a primary key... it needs to be possible for a patient to have more than one medication order!
+
+</div>
+
+***********
+
+Consider the same table.
+
+Which of the fields in this table is likely to be a foreign key?  Select all that apply.
+
+[[ ]] med\_ord\_id
+[[X]] patient\_id
+[[ ]] date
+[[X]] provider\_id
+*********
+
+<div class = "answer">
+
+A foreign key is an identifier that originates in another table.  It certainly seems likely that there is a "patient" table and a "provider" table that the "patient\_id" and "provider\_id", respectively, originated in.  These are the two foreign keys.  The "med\_ord\_id" seems to originate here, in the "medication_order" table, so it's not a foreign key, and the "date" just seems like, well, a date, not an identifier at all!
```

Comment:
Question: are there situations in which a date might be a primary key for say research on whether patients have better or worse outcomes on holidays or when new residents start in the summer? Or on rainy days?

This might be pushing too far, but I'm genuinely curious.

Reply:
yep, I could envision that, it's possible, but it's an edge case.